File: backend/src/db-queries.ts
import { db, schema } from './db/index.js';
import { eq } from 'drizzle-orm';
import { StreamConfig } from './types.js';

// Get a stream by ID
export async function getStreamById(streamId: string): Promise<StreamConfig | null> {
  try {
    const id = parseInt(streamId, 10);
    if (isNaN(id)) {
      return null;
    }
    
    const result = await db
      .select({
        id: schema.streams.id,
        url: schema.streams.url,
      })
      .from(schema.streams)
      .where(eq(schema.streams.id, id))
      .limit(1);
    
    if (result.length === 0) {
      return null;
    }
    
    return {
      id: result[0].id.toString(),
      url: result[0].url,
    };
  } catch (error) {
    console.error('Error getting stream:', error);
    throw error;
  }
}

// Create a new stream
export async function createStream(url: string): Promise<StreamConfig> {
  try {
    const result = await db.insert(schema.streams).values({
      url: url,
    }).returning({
      id: schema.streams.id,
      url: schema.streams.url,
    });
    
    return {
      id: result[0].id.toString(),
      url: result[0].url,
    };
  } catch (error) {
    console.error('Error creating stream:', error);
    throw error;
  }
}

File: backend/src/db_queries.py
from typing import Optional, Dict
from sqlmodel import select
from .neon_db import get_session
from .models import Stream


async def get_stream_by_id(stream_id: str) -> Optional[Dict[str, str]]:
    """Get a stream by ID"""
    try:
        stream_id_int = int(stream_id)
    except ValueError:
        return None
    
    async with get_session() as session:
        statement = select(Stream).where(Stream.id == stream_id_int)
        result = await session.execute(statement)
        stream = result.scalar_one_or_none()
        
        if not stream:
            return None
        
        return {
            "id": str(stream.id),
            "url": stream.url,
        }


async def create_stream(url: str) -> Dict[str, str]:
    """Create a new stream"""
    async with get_session() as session:
        stream = Stream(url=url)
        session.add(stream)
        await session.commit()
        await session.refresh(stream)
        
        return {
            "id": str(stream.id),
            "url": stream.url,
        }

